srv/modules/runners/osd.py

```diff
@@ -410,13 +408,44 @@ def pre_check(osd_list, force):
         ok_to_stop_osds(osd_list)
 
 
+def _target_lookup(inp):
+    """
+    This allows to specify a compound taret on the commandline
+    """
+    osd_list = []
+    osd_list_string = ''
+    for target in inp:
+        if isinstance(target, int):
+            # It's safe to assume that when the first arg is
+            # an integer we get a list of osd_ids
+            return list(inp), " ".join([str(x) for x in list(inp)])
+        elif isinstance(target, str):
+            print("Detected a compound target")
+            osd_list_return = salt.client.LocalClient().cmd(
+                target, "osd.list", tgt_type="compound")
+
+            if not osd_list_return:
+                print("Target: {} did not match anything".format(target))
+                return False
+            for _, osds in osd_list_return.items():
+                if osds:
+                    osd_list.extend(osds)
+            # ceph expects a string
+            osd_list_string = " ".join([str(x) for x in osd_list])
+            print("Found OSDS: {} in compound target {}".format(
+                osd_list, target))
+    return osd_list, osd_list_string
+
+
 def remove(*args, **kwargs):
     """ User facing osd.remove function """
     results = dict()
     kwargs.update({'operation': 'remove'})
-    osd_list = " ".join([str(x) for x in list(args)])
-    pre_check(osd_list, kwargs.get('force', False))
-    for osd_id in args:
+    osd_list, osd_list_string = _target_lookup(args)
+    if not osd_list:
+        return False
+    pre_check(osd_list_string, kwargs.get('force', False))
```

Comment:
Rather than having `_target_lookup` return both a list and a string, it might simplify things to have it return only a list, then `.join` that into a string here, which seems to be the only place the string is needed.

Reply:
Right, I have a comment about that somewhere in my notes.. I think it'd be even better to move the list->str conversion `pre_check`. I'll update this shortly